pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -902,6 +925,61 @@ func dnsZoneName(cdName string) string {
 	return fmt.Sprintf("%s-zone", cdName)
 }
 
+func selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
```

Comment:
just a style suggestion (feel free to ignore) -- exit early here
if pod.Labels == nil {
     return retval
}